x/claim/keeper/activity.go

```diff
@@ -0,0 +1,89 @@
+package keeper
+
+import (
+	"encoding/json"
+
+	"github.com/c-osmosis/osmosis/x/claim/types"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// GetActivities get activites of users for genesis export
+func (k Keeper) GetActivities(ctx sdk.Context) []types.UserActivities {
+	store := ctx.KVStore(k.storeKey)
+	prefixStore := prefix.NewStore(store, append([]byte(types.ActionKey)))
+	iterator := prefixStore.Iterator(nil, nil)
+	defer iterator.Close()
+
+	actionsByUser := make(map[string]types.Actions)
+	for ; iterator.Valid(); iterator.Next() {
+		value := types.UserActivity{}
+		err := json.Unmarshal(iterator.Value(), &value)
+		if err != nil {
+			panic(err)
+		}
+		actions := actionsByUser[value.User]
+		actions = append(actions, value.Action)
+		actionsByUser[value.User] = actions
+	}
+
+	activities := []types.UserActivities{}
+	for user, actions := range actionsByUser {
+		activities = append(activities, types.UserActivities{
+			User:      user,
+			Actions:   actions,
+		})
+	}
+	return activities
+}
+
+func (k Keeper) SetUserActions(ctx sdk.Context, address sdk.AccAddress, actions []types.Action) {
+	for _, action := range actions {
+		k.SetUserAction(ctx, address, action)
+	}
+}
+
+func (k Keeper) GetUserActions(ctx sdk.Context, address sdk.AccAddress) []types.Action {
+	store := ctx.KVStore(k.storeKey)
+	prefixStore := prefix.NewStore(store, append([]byte(types.ActionKey), address.Bytes()...))
+	iterator := prefixStore.Iterator(nil, nil)
+	defer iterator.Close()
+
+	actions := []types.Action{}
+	for ; iterator.Valid(); iterator.Next() {
+		value := types.UserActivity{}
+		err := json.Unmarshal(iterator.Value(), &value)
+		if err != nil {
+			panic(err)
+		}
+		actions = append(actions, value.Action)
+	}
+
+	return actions
+}
+
+func (k Keeper) SetUserAction(ctx sdk.Context, address sdk.AccAddress, action types.Action) bool {
+	store := ctx.KVStore(k.storeKey)
+	prefixStore := prefix.NewStore(store, []byte(types.ActionKey))
+	key := append(address, sdk.Uint64ToBigEndian(uint64(action))...)
+	if prefixStore.Has(key) {
+		return false
+	}
+	value := types.UserActivity{
+		User:   address.String(),
+		Action: action,
+	}
+	valueBz, err := json.Marshal(value)
+	if err != nil {
+		panic(err)
+	}
+	prefixStore.Set(key, valueBz)
+	return true
+}
+
+// GetClaimablePercentageByActivity returns percentage by user's activity when the weight of actions are same
+func (k Keeper) GetClaimablePercentageByActivity(ctx sdk.Context, address sdk.AccAddress) sdk.Dec {
+	numTotalActions := len(types.Action_name)
+	return sdk.NewDec(1).QuoInt64(int64(numTotalActions))
```

Comment:
Should this be adjusted to account for the `1` osmo given unclaimed? 
/cc @sunnya97

Reply:
EDIT: Just saw that this is called from ClaimCoins, so it must account for this

Reply:
1 Osmo is nothing I think - user is using it only for fee.